source/MRMesh/MRPolylineTrimWithPlane.cpp

```diff
@@ -5,175 +5,217 @@
 
 namespace MR
 {
-    EdgeBitSet subdividePolylineWithPlane( Polyline3& polyline, const Plane3f& plane, std::function<void( EdgeId, EdgeId, float )> onEdgeSplitCallback )
-    {
-        if ( polyline.topology.numValidVerts() == 0 )
-            return {};
 
-        EdgeBitSet res;
-        const auto sectionPoints = extractSectionsFromPolyline( polyline, plane, 0.0f );
-        for ( const auto& sectionPoint : sectionPoints )
-        {
-            const auto eNew = polyline.splitEdge( sectionPoint.e, polyline.edgePoint( sectionPoint.edgePointA() ) );            
-            res.autoResizeSet( sectionPoint.e );
-            if ( onEdgeSplitCallback )
-                onEdgeSplitCallback( sectionPoint.e, eNew, sectionPoint.a );
-        }
+EdgeBitSet subdivideWithPlane( Polyline3& polyline, const Plane3f& plane, EdgeBitSet* newPositiveEdges, std::function<void( EdgeId, EdgeId, float )> onEdgeSplitCallback )
+{
+    if ( polyline.topology.numValidVerts() == 0 )
+        return {};
 
-        return res;
-    }
+    if ( newPositiveEdges )
+        *newPositiveEdges = {};
```

Comment:
It is ok to remove it and add the comment that we only add bits in given `newPositiveEdges`

Reply:
fixed